service/common/src/main/java/org/apache/polaris/service/catalog/generic/GenericTableCatalogHandler.java

```diff
@@ -77,7 +77,7 @@ public ListGenericTablesResponse listGenericTables(Namespace parent) {
     authorizeBasicNamespaceOperationOrThrow(op, parent);
 
     return ListGenericTablesResponse.builder()
-        .setIdentifiers(new TreeSet<>(genericTableCatalog.listGenericTables(parent)))
+        .setIdentifiers(new LinkedHashSet<>(genericTableCatalog.listGenericTables(parent)))
```

Comment:
We should prefer to preserve the existing order, not necessarily to sort by name